src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionAuthorizationRequestedQueueConsumer.kt

```diff
@@ -0,0 +1,79 @@
+package it.pagopa.ecommerce.eventdispatcher.queues
+
+import com.azure.core.util.BinaryData
+import com.azure.core.util.serializer.TypeReference
+import com.azure.spring.messaging.AzureHeaders
+import com.azure.spring.messaging.checkpoint.Checkpointer
+import it.pagopa.ecommerce.commons.documents.v2.TransactionAuthorizationRequestedEvent as TransactionAuthorizationRequestedEventV2
+import it.pagopa.ecommerce.commons.queues.QueueEvent
+import it.pagopa.ecommerce.commons.queues.StrictJsonSerializerProvider
+import it.pagopa.ecommerce.commons.queues.TracingInfo
+import it.pagopa.ecommerce.eventdispatcher.exceptions.InvalidEventException
+import it.pagopa.ecommerce.eventdispatcher.utils.DeadLetterTracedQueueAsyncClient
+import java.util.*
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.integration.annotation.ServiceActivator
+import org.springframework.messaging.handler.annotation.Header
+import org.springframework.messaging.handler.annotation.Payload
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+
+/**
+ * Event consumer for transactions for which retrieve authorization outcome status. This consumer
+ * will handle TransactionAuthorizationRequestedEvent V2 events and inquiry gateway in order to
+ * retrieve authorization outcome
+ */
+@Service("TransactionAuthorizationRequestedQueueConsumer")
+class TransactionAuthorizationRequestedQueueConsumer(
+  @Autowired
+  private val queueConsumerV2:
+    it.pagopa.ecommerce.eventdispatcher.queues.v2.TransactionAuthorizationRequestedQueueConsumer,
+  @Autowired private val deadLetterTracedQueueAsyncClient: DeadLetterTracedQueueAsyncClient,
+  @Autowired private val strictSerializerProviderV2: StrictJsonSerializerProvider
+) {
+
+  var logger: Logger =
+    LoggerFactory.getLogger(TransactionAuthorizationRequestedQueueConsumer::class.java)
+
+  fun parseEvent(
+    payload: ByteArray
+  ): Mono<Pair<TransactionAuthorizationRequestedEventV2, TracingInfo?>> {
```

Comment:
```suggestion
  ): Mono<QueueEvent<TransactionAuthorizationRequestedEventV2>> {
```

make this function return queue event directly so that it can be used as is in the below code instead of building a new QueueEvent from the exploded parsed event